### PR TITLE
[Workflows] Add `reviewdog` workflow and `PR` template

### PR DESCRIPTION
**Pull request by Olshansk**
- Add a `PR` template
- Add a `reviewdog` template in CI

### CODE REVIEW

File: .github/workflows/label-actions.yml [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+  issues: write
+  pull-requests: write
+  discussions: write
+
+jobs:
+  action:
+    runs-on: ubuntu-latest
+    steps:
+      # TODO_TECHDEBT: switch to `dessant/label-actions@v3` when https://github.com/dessant/label-actions/pull/29 is merged.
+      - uses: okdas/label-actions@patch-1
```
Comment (okdas) [inline .github/workflows/label-actions.yml:22]: What kind of actions do you want this workflow to perform? It requires another file: `.github/label-actions.yml` - https://github.com/pokt-network/poktroll/blob/main/.github/label-actions.yml - which contains the different actions that can happen when the label is present.

**Review by okdas — CHANGES_REQUESTED** [review]
Thank you for adding those, we just need another file transferred and modified for the workflow to work.
If you tell me what you want - I can do it. In `poktroll` repo, we use this workflow to let the developer know some gotchas or provide links for devnets.

File: .github/workflows/label-actions.yml (hunk 1 above)
Comment (Olshansk) [reply .github/workflows/label-actions.yml:22]: Noted. This is good insight. My goal was to get `In Review` to automatically switch to `Done` once merge in the PR. 

I couldn't find a GitHub setting for it, so I assume this is the  tool to make that happen.

Do you know if that's a default elsewhere?

Comment (okdas) [reply .github/workflows/label-actions.yml:22]: @Olshansk  I was actually curious how this works and this is what I found: https://github.com/orgs/pokt-network/projects/144/workflows

Apparently this is a new feature github released earlier this year: https://github.blog/changelog/2024-04-25-github-issues-projects-auto-close-issue-project-workflow/

Comment (Olshansk) [reply .github/workflows/label-actions.yml:22]: Interesting!

I just added a few of them to our grove project as well: https://github.com/orgs/buildwithgrove/projects/1/workflows/32463471

**Comment by Olshansk** [comment]
@okdas Changed the goal of this PR so please take a look again when you can.

**Comment by Olshansk** [comment]
@okdas Friendly bump on this please

**Review by okdas — APPROVED** [review]
Looks good!

**Comment by Olshansk** [comment]
I'm bypassing E2E tests introduced in #103 because these are just github workflow.

**Pull request merged**